go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/submariner-io/submariner-operator
 
-go 1.21
+go 1.23.0
```

Comment:
DO we want to bump the Go version? This job failed: https://github.com/submariner-io/submariner-operator/actions/runs/14646863521/job/41103156368?pr=3430